WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationTracker.kt

```diff
@@ -4,9 +4,114 @@ import org.wordpress.android.analytics.AnalyticsTracker
 import javax.inject.Inject
 import javax.inject.Singleton
 
+const val ORIGIN_VERTICALS_FULLSCREEN_ERROR = "Verticals - fullscreen"
+const val ORIGIN_VERTICALS_LIST_ITEM_ERROR = "Verticals - list item"
+const val ORIGIN_SEGMENTS_ERROR = "Segments"
+const val ORIGIN_DOMAINS_ERROR = "Domains"
+const val ORIGIN_PREVIEW_ERROR = "Preview"
+
 @Singleton
 class NewSiteCreationTracker @Inject constructor() {
-    fun trackSiteCreated() {
-        AnalyticsTracker.track(AnalyticsTracker.Stat.CREATED_SITE)
+    fun trackSiteCreationAccessed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_ACCESSED)
+    }
+
+    fun trackSegmentsViewed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_SEGMENTS_VIEWED)
+    }
+
+    fun trackSegmentSelected(segmentName: String, segmentId: Long) {
+        AnalyticsTracker.track(
+                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_SEGMENTS_SELECTED,
+                mapOf(
+                        "segment_name" to segmentName,
+                        "segment_id" to segmentId
+                )
+        )
+    }
+
+    fun trackVerticalsViewed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_VERTICALS_VIEWED)
+    }
+
+    fun trackVerticalSelected(verticalName: String, verticalId: String, isUserVertical: Boolean) {
+        AnalyticsTracker.track(
+                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_VERTICALS_SELECTED,
+                mapOf(
+                        "vertical_name" to verticalName,
+                        "vertical_id" to verticalId,
+                        "vertical_is_user" to isUserVertical
+                )
+        )
+    }
+
+    fun trackVerticalsSkipped() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_VERTICALS_SKIPPED)
+    }
+
+    fun trackBasicInformationViewed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_BASIC_INFORMATION_VIEWED)
+    }
+
+    fun trackBasicInformationCompleted(siteTitle: String, siteTagLine: String) {
+        AnalyticsTracker.track(
+                AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_BASIC_INFORMATION_COMPLETED,
+                mapOf(
+                        "site_title" to siteTitle,
+                        "tagline" to siteTagLine
+                )
+        )
+    }
+
+    fun trackBasicInformationSkipped() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_BASIC_INFORMATION_SKIPPED)
+    }
+
+    fun trackDomainsAccessed() {
+        AnalyticsTracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_DOMAINS_ACCESSED)
+    }
+
+    fun trackDomainSelected(choosenDomain: String, searchTerm: String) {
```

Comment:
Nitpick: There is a small typo here.

```suggestion
    fun trackDomainSelected(chosenDomain: String, searchTerm: String) {
```